File: src/AnalyticsService/Transport/Validation/BatchStatRequestValidator.fs
namespace AnalyticsService.Transport.Validation

open AnalyticsService.Transport.Contracts.Requests
open FluentValidation

/// A validator class for BatchStatRequest record.
[<Sealed>]
type BatchStatRequestValidator() as this =
    inherit AbstractValidator<BatchStatRequest>()
    do
        this.RuleFor(fun i -> i.StartDate).NotEmpty() |> ignore
        this
            .RuleFor(fun i -> i.EndDate)
            .NotEmpty()
            .GreaterThan(fun i -> i.StartDate) |> ignore
        this.RuleFor(fun i -> i.Status).NotEmpty() |> ignore
        this.RuleFor(fun i -> i.NumberOfDocuments).GreaterThan(0) |> ignore

File: src/AnalyticsService/Transport/Validation/BatchStatRequestValidator.cs
using AnalyticsService.Transport.Contracts.Requests;
using FluentValidation;

namespace AnalyticsService.Transport.Validation;

/// <summary>
/// A validator class for <see cref="BatchStatRequest"/> record.
/// </summary>
internal sealed class BatchStatRequestValidator : AbstractValidator<BatchStatRequest>
{
    public BatchStatRequestValidator()
    {
        RuleFor(i => i.StartDate).NotEmpty();
        RuleFor(i => i.EndDate).NotEmpty().GreaterThan(i => i.StartDate);
        RuleFor(i => i.Status).IsInEnum();
        RuleFor(i => i.NumberOfDocuments).GreaterThan(0);
    }
}